ibis/expr/datatypes.py

```diff
@@ -946,6 +1043,28 @@ def type(self) -> DataType:
         struct : "struct" "<" field ":" type ("," field ":" type)* ">"
 
         field : [a-zA-Z_][a-zA-Z_0-9]*
+
```

Comment:
@xmnlab please add test cases for the newly introduces parsing parts

Reply:
So for example add extend the parameters of [test_dtype](https://github.com/ibis-project/ibis/blob/master/ibis/expr/tests/test_datatypes.py#L27) with `linestring;<srid>:<geotype>`

Reply:
@xmnlab look the uncovered lines: https://codecov.io/gh/ibis-project/ibis/pull/1666/src/ibis/expr/datatypes.py#L1191

Reply:
Codecov doesn't redirect properly, see the lines from 1191.

Reply:
Tests added

Reply:
thanks @kszucs! let me know if it is missing anything else :+1: